### DISCUSSION TITLE
React Native OnlineManager Weirdness

### DISCUSSION THREAD

**Discussion by wilkinson4** [TanStack/query#5429] [Q&A]
Hello, maybe something is weird in my android emulator, but I have the recommended `onlineManager` config in my App.js file:
```js
const App = () => {
  
  onlineManager.setEventListener(setOnline => {
    return addEventListener(state => {
      setOnline(!!state.isConnected);
    });
  });
  return (
   ...
  )
}
```

What I'm noticing is that if I go from online to airplane mode, try to make a request, then turn airplane mode off it makes the request as expected when the device reconnects. However, if I repeat this behaviour, it never fires the request, but it does capture the network status update. I know this because I have an `OfflineIndicator` component that is using `onlineManager.isOnline()` to conditionally display an offline message, and that seems to be working fine.

I'm on the latest version of react-query, running react-native version 0.71.3, and react version 18.2.0. The app that I'm working on is only for Android. 

Any help would be greatly appreciated!

**Comment by amit-maitri** [comment]
Hey there's a comment in the docs mentioning `addEventListener` doesn't exist in React Native, the code is using `window.addEventListener`

check it out: https://github.com/TanStack/query/blob/1047cdc393fac7c98822c993d70c28f58833c63d/packages/query-core/src/onlineManager.ts#L15
